test/conformance/source/environment.cpp

```diff
@@ -199,6 +201,14 @@ DevicesEnvironment::DevicesEnvironment(int argc, char **argv)
         error = "Could not find any devices associated with the platform";
         return;
     }
+    // Get a parameter (NUMBER_OF_DEVICES) to run tests on the appropriate number of devices
+    auto devices_num = ur_getenv("NUMBER_OF_DEVICES");
+    uint32_t count_set = devices_num ? static_cast<uint32_t>(std::stoi(devices_num.value())) : 1;
+    if (count_set <= 0 || count_set > std::numeric_limits<uint32_t>::max()) {
```

Comment:
Doing this check after casting the value to `uint32_t` is pointless. You should do this check before the cast.

Reply:
Done

Reply:
if we parse string already to proper integer value (uint32) we may omit this check as it will throw (out_of_range ex)

Reply:
I haven't found such a function. I would leave it like that.